lab-sarah/ng-template/test/gallery-li-controller-test.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+describe('testing gallery-li controller', function(){
+  beforeEach(() => {
+    angular.mock.module('demoApp');
+    angular.mock.inject(($rootScope, $componentController, authService, $httpBackend) => {
+      authService.setToken('secrettoken');
+      this.authService = authService;
+      this.$httpBackend = $httpBackend;
+      this.$rootScope = $rootScope;
+      this.$componentController = $componentController;
+    });
+  });
+
+  afterEach( () => {
+    //for expect statements
+    this.$httpBackend.verifyNoOutstandingExpectation();
+    //for when statements
+    this.$httpBackend.verifyNoOutstandingRequest();
+  });
+
+  afterEach(() => {
+    this.authService.logout();
+  });
+
+  describe('testing #deleteDone', () => {
+    it('should call deleteDone', () => {
+      let mockBindings = {
+        gallery: {
+          _id: '65432ONE',
+          name: 'hello',
+          desc: 'infomative',
+          pics: [],
+        },
+        deleteDone: function(data){
+          expect(data.galleryData._id).toEqual('65432ONE');
+        },
+      };
+
+      let galleryLICtrl = this.$componentController('galleryLi', null, mockBindings);
+      //now call the deleteDone method on the controller, and pass in the.....gallery
+      galleryLICtrl.deleteDone({galleryData: galleryLICtrl.gallery});
+
+      this.$rootScope.$apply();
+    });
+  });
+
+  it('should call deleteDone with gallery after galleryDelete', () => {
+    let url = 'http://localhost:3000/api/gallery/65432ONE';
+    let headers = {
+      Authorization: 'Bearer secrettoken',
+      Accept: 'application/json, text/plain, */*',
+    };
+
+    let mockBindings = {
+      gallery: {
+        _id: '65432ONE',
+        name: 'hello',
+        desc: 'infomative',
+        pics: [],
+      },
+      deleteDone: function(data){
+        expect(data.galleryData._id).toEqual(mockBindings.gallery._id);
+      },
+    };
+
+    this.$httpBackend.expectDELETE(url, headers).respond(204);
+
+    let galleryLICtrl = this.$componentController('galleryLi', null, mockBindings);
+    galleryLICtrl.deleteGallery();
```

Comment:
Same as above -- you are not testing the method itself here. You currently can't really do this without resolving something from your galleryLICtrl. Let me know if you need clarification on this.